### PR TITLE
Remove TOTAL_MEMORY global var from emitted JS

### PR DESCRIPTION
**Pull request by kripken**
(This does not affect the TOTAL_MEMORY setting value.)

`TOTAL_MEMORY` just mirrors e.g. `HEAP8.length`, so it's some state that can get out of sync. It is in fact hard to keep it in sync for pthreads, as each worker has an instance. Instead, just directly read `HEAP8.length` when we need the size of memory.

### CODE REVIEW

File: tests/core/test_memorygrowth_wasm_mem_max.c [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 int main() {
   const int MB = 1024 * 1024;
-  // TOTAL_MEMORY starts at 64MB, and max is 100. allocate enough
+  // Total memory starts at 64MB, and max is 100. allocate enough
```
Comment (sbc100) [inline tests/core/test_memorygrowth_wasm_mem_max.c]: Memory size starts at ..?

File: tests/mallinfo.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ extern "C" {
 
 unsigned int getTotalMemory()
 {
-	return EM_ASM_INT(return TOTAL_MEMORY);
```
Comment (sbc100) [inline tests/mallinfo.cpp]: Technically this can be breaking change if any user code does this kind of thing, right?

Comment (kripken) [reply tests/mallinfo.cpp]: It is, yeah - but this is an internal implementation detail we've never documented as stable (or documented at all, I think ;)

File: tests/core/test_memorygrowth_wasm_mem_max.c (hunk 1 above)
Comment (kripken) [reply tests/core/test_memorygrowth_wasm_mem_max.c]: Fixing, thanks.

**Pull request merged**